lib/sycamore/sycamore/evaluation/evaluate.py

```diff
@@ -0,0 +1,127 @@
+from sycamore import Context
+
+
+class Assessment(ABC):
+
+    @abstractmethod
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        pass
+
+    def __call__(self, context: Context, index: str, **kwargs):
+        return self.run_evaluation(context, index, **kwargs)
+
+
+class QualityAssessment(Assessment):
+    def __init__(self, GT_path: str, rag_config: Dict, **kwargs):
+        self.user = os.environ.get("USER", os.environ.get("USERNAME"))
+        self.GT_path = GT_path
+        self.rag_config = rag_config
+        self.os_client_args = kwargs.get("os_client_args", "")
+        self.metrics = kwargs.get("metrics", [document_retrieval_metrics, rouge_metrics])
+        self.custom_question_augmentation = kwargs.get("custom_question_augmentation", {})
+        self.question_augmentation_filter = kwargs.get("question_augmentation_filter", "")
+
+    @staticmethod
+    def create_evaluation_datapoint(
+        json_dict: Dict, custom_question_augmentation: str = "{}", question_augmentation_filter: str = ""
+    ):
+        result = []
+        assert json_dict is not None
+        assert isinstance(json_dict, dict)
+        for datapoint in json_dict["data"][:]:
+            document = EvaluationDataPoint()
+            document.raw = datapoint
+            document.ground_truth_answer = datapoint["Answer"]
+            document.filters = datapoint.get("Filters", None)
+            if document.filters:
+                for filter in document.filters.keys():
+                    document.filters[filter] = datapoint.get("Filters").get(filter)
+                document.question = custom_question_augmentation.format(
+                    document.question, document.filters.get(question_augmentation_filter)
+                )
+            else:
+                document.filters = {}
+                document.question = custom_question_augmentation.format(document.question, "")
+            source_documents: List[Element] = []
+            for search_result in datapoint["SearchContexts"]:
+                source_document = Element()
+                properties = {
+                    "_location": search_result["document_url"],
+                    "page_number": search_result["page_numbers"][0],
+                    "doc_id": search_result["document_id"],
+                }
+                source_document.properties = properties
+                source_document.text_representation = search_result["text_representation"]
+                source_documents += [source_document]
+
+            document.ground_truth_source_documents = source_documents
+            result += [{"doc": document.serialize()}]
+        return result
+
+    def run_evaluation(self, ctx: Context, index: str, **kwargs):
+        custom_question_augmentation = str(self.custom_question_augmentation)
+        question_augmentation_filter = str(self.question_augmentation_filter)
+        input_docset = DocSetReader(ctx).json(
```

Comment:
`ctx.read.json(...)` is a little cleaner and saves you an import

Reply:
this definitely is more cleaner, change made !